.github/workflows/appetize.yml

```
@@ -0,0 +1,162 @@
+        run: |
+          curl --silent -o appetize.json https://$APPETIZE_TOKEN@api.appetize.io/v1/apps/$APPETIZE_APP > /dev/null
+          echo "Appetize is running $(cat appetize.json | jq .appVersionName)"
+          rm -rf appetize.json
+        env:
+          APPETIZE_TOKEN: ${{ secrets.APPETIZE_MAIN_TOKEN }}
+          APPETIZE_APP: ${{ secrets.APPETIZE_MAIN_ANDROID }}
+      
+      - name: Resolve Appetize (embed) version
+        if: contains('all embed', github.event.inputs.appetizeQueue)
+        # This step MUST only output `appVersionName`, the endpoint returns the public and private app keys
+        run: |
+          curl --silent -o appetize.json https://$APPETIZE_TOKEN@api.appetize.io/v1/apps/$APPETIZE_APP > /dev/null
+          echo "Appetize is running $(cat appetize.json | jq .appVersionName)"
+          rm -rf appetize.json
+        env:
+          APPETIZE_TOKEN: ${{ secrets.APPETIZE_EMBED_TOKEN }}
+          APPETIZE_APP: ${{ secrets.APPETIZE_EMBED_ANDROID }}
+
+      - name: Resolve Android client
+        if: github.event.inputs.sdkVersion
+        id: android
+        run: |
+          curl --silent -o versions.json https://exp.host/--/api/v2/versions
+
+          VERSION=${{ github.event.inputs.sdkVersion }}
+          CLIENT_URL=$(cat versions.json | jq .sdkVersions[\"$VERSION\"].androidClientUrl)
+          CLIENT_VERSION=$(cat versions.json | jq .sdkVersions[\"$VERSION\"].androidClientVersion)
+          
+          if [ "${CLIENT_URL}" == "null" ]; then echo "No client url for SDK '$VERSION'"; exit 1; fi
+          if [ "${CLIENT_VERSION}" == "null" ]; then echo "No client version for SDK '$VERSION'"; exit 1; fi
+
+          echo "Resolved client $CLIENT_VERSION from SDK $VERSION"
+          echo "::set-output name=url::$CLIENT_URL"
+          echo "::set-output name=version::$CLIENT_VERSION"
+      
+      - name: Download Android client
+        if: github.event.inputs.deploy == 'deploy' && github.event.inputs.sdkVersion
```

Using `deploy` without filling in the `sdkVersion` will just skip this step. We should probably throw an error when using `deploy` while `sdkVersion` is not specified

---

Yeah, maybe just erroring out might be better. I did this to avoid running this step because it actually did go off when SDK version is empty.

---

Done in the revised version #195